roachtest: sqlsmith/setup=seed/setting=no-ddl failed

[(roachtest).sqlsmith/setup=seed/setting=no-ddl failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2143445&tab=buildLog) on [provisional_202008031850_v20.2.0-alpha.3@b0ec3c9834d59be51e7af8a070bbef8b20f2d62a](https://github.com/cockroachdb/cockroach/commits/b0ec3c9834d59be51e7af8a070bbef8b20f2d62a):

```
		INSERT
		INTO
			defaultdb.public.seed AS tab_2421
		WITH
			with_798 (col_6280) AS (SELECT * FROM (VALUES ((-1.569702273301134):::FLOAT8)) AS tab_2422 (col_6280))
		SELECT
			4917:::INT8 AS col_6281,
			(
				SELECT
					(-319347211):::INT8 AS col_6282
				FROM
					defaultdb.public.seed@seed__int8__float8__date_idx AS tab_2423, defaultdb.public.seed@[0] AS tab_2424
				WHERE
					false
				ORDER BY
					tab_2424._float8 ASC, tab_2424._float4 ASC, tab_2423._timestamptz
				LIMIT
					1:::INT8
			)
				AS col_6283,
			crdb_internal.destroy_tenant((-3469236868022851591):::INT8::INT8)::INT8 AS col_6284,
			cte_ref_218.col_6280 AS col_6285,
			var_pop(cte_ref_218.col_6280::FLOAT8) OVER (PARTITION BY cte_ref_218.col_6280 RANGE CURRENT ROW)::FLOAT8
				AS col_6286,
			now():::DATE::DATE AS col_6287,
			current_timestamp((-5367689394412957846):::INT8::INT8):::TIMESTAMP::TIMESTAMP AS col_6288,
			experimental_follower_read_timestamp()::TIMESTAMPTZ AS col_6289,
			date_trunc(e'dSwj\rx^p*':::STRING::STRING, '11:11:22.724072':::TIME::TIME)::INTERVAL AS col_6290,
			st_covers('010500000009000000010200000002000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000004000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000005000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000002000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000002000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000004000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000002000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000006000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000004000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F':::GEOMETRY::GEOMETRY, '0101000000000000000000F07F000000000000F07F':::GEOMETRY::GEOMETRY)::BOOL
				AS col_6291,
			exp((-1):::DECIMAL::DECIMAL)::DECIMAL AS col_6292,
			overlay('K !cc':::STRING::STRING, NULL::STRING, NULL::INT8)::STRING AS col_6293,
			crdb_internal.encode_key(NULL::INT8, NULL::INT8, B'0111111111111111111111111111111111111111111111111111111111111111')::BYTES
				AS col_6294,
			gen_random_uuid()::UUID AS col_6295,
			netmask(broadcast(NULL::INET)::INET::INET)::INET AS col_6296,
			json_agg(cte_ref_218.col_6280) OVER (PARTITION BY cte_ref_218.col_6280 RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING)::JSONB
				AS col_6297,
			'hello':::greeting AS col_6298,
			pow((-7.92966261239411176E+31):::DECIMAL::DECIMAL, (-90438879243.33829799):::DECIMAL::DECIMAL)::DECIMAL AS col_6299
		FROM
			with_798 AS cte_ref_218
		WHERE
			true
		GROUP BY
			cte_ref_218.col_6280
		HAVING
			st_covers('X':::STRING::STRING, st_ashexewkb('010200000004000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F':::GEOMETRY::GEOMETRY)::STRING::STRING)::BOOL
		ORDER BY
			cte_ref_218.col_6280 DESC, cte_ref_218.col_6280 DESC;

```

<details><summary>More</summary><p>


Artifacts: [/sqlsmith/setup=seed/setting=no-ddl](https://teamcity.cockroachdb.com/viewLog.html?buildId=2143445&tab=artifacts#/sqlsmith/setup=seed/setting=no-ddl)
Related:
- #51313 roachtest: sqlsmith/setup=seed/setting=no-ddl failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202007081918_v20.2.0-alpha.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202007081918_v20.2.0-alpha.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #48269 roachtest: sqlsmith/setup=seed/setting=no-ddl failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-master](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-master) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Asqlsmith%2Fsetup%3Dseed%2Fsetting%3Dno-ddl.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

```
	sqlsmith.go:172,sqlsmith.go:202,test_runner.go:754: error: pq: internal error: interface conversion: tree.Datum is *tree.DString, not *tree.DGeometry
		stmt:
		INSERT
		INTO
			defaultdb.public.seed AS tab_2421
		WITH
			with_798 (col_6280) AS (SELECT * FROM (VALUES ((-1.569702273301134):::FLOAT8)) AS tab_2422 (col_6280))
		SELECT
			4917:::INT8 AS col_6281,
			(
				SELECT
					(-319347211):::INT8 AS col_6282
				FROM
					defaultdb.public.seed@seed__int8__float8__date_idx AS tab_2423, defaultdb.public.seed@[0] AS tab_2424
				WHERE
					false
				ORDER BY
					tab_2424._float8 ASC, tab_2424._float4 ASC, tab_2423._timestamptz
				LIMIT
					1:::INT8
			)
				AS col_6283,
			crdb_internal.destroy_tenant((-3469236868022851591):::INT8::INT8)::INT8 AS col_6284,
			cte_ref_218.col_6280 AS col_6285,
			var_pop(cte_ref_218.col_6280::FLOAT8) OVER (PARTITION BY cte_ref_218.col_6280 RANGE CURRENT ROW)::FLOAT8
				AS col_6286,
			now():::DATE::DATE AS col_6287,
			current_timestamp((-5367689394412957846):::INT8::INT8):::TIMESTAMP::TIMESTAMP AS col_6288,
			experimental_follower_read_timestamp()::TIMESTAMPTZ AS col_6289,
			date_trunc(e'dSwj\rx^p*':::STRING::STRING, '11:11:22.724072':::TIME::TIME)::INTERVAL AS col_6290,
			st_covers('010500000009000000010200000002000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000004000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000005000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000002000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000002000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000004000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000002000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000006000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F010200000004000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F':::GEOMETRY::GEOMETRY, '0101000000000000000000F07F000000000000F07F':::GEOMETRY::GEOMETRY)::BOOL
				AS col_6291,
			exp((-1):::DECIMAL::DECIMAL)::DECIMAL AS col_6292,
			overlay('K !cc':::STRING::STRING, NULL::STRING, NULL::INT8)::STRING AS col_6293,
			crdb_internal.encode_key(NULL::INT8, NULL::INT8, B'0111111111111111111111111111111111111111111111111111111111111111')::BYTES
				AS col_6294,
			gen_random_uuid()::UUID AS col_6295,
			netmask(broadcast(NULL::INET)::INET::INET)::INET AS col_6296,
			json_agg(cte_ref_218.col_6280) OVER (PARTITION BY cte_ref_218.col_6280 RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING)::JSONB
				AS col_6297,
			'hello':::greeting AS col_6298,
			pow((-7.92966261239411176E+31):::DECIMAL::DECIMAL, (-90438879243.33829799):::DECIMAL::DECIMAL)::DECIMAL AS col_6299
		FROM
			with_798 AS cte_ref_218
		WHERE
			true
		GROUP BY
			cte_ref_218.col_6280
		HAVING
			st_covers('X':::STRING::STRING, st_ashexewkb('010200000004000000000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F000000000000F07F':::GEOMETRY::GEOMETRY)::STRING::STRING)::BOOL
		ORDER BY
			cte_ref_218.col_6280 DESC, cte_ref_218.col_6280 DESC;
```
cc @otan

--- a/pkg/geo/geogen/geogen.go
+++ b/pkg/geo/geogen/geogen.go
@@ -185,8 +185,8 @@ func RandomGeomT(
 
 // RandomGeometry generates a random Geometry with the given SRID.
 func RandomGeometry(rng *rand.Rand, srid geopb.SRID) *geo.Geometry {
-	minX, maxX := -math.MaxFloat64, math.MaxFloat64
-	minY, maxY := -math.MaxFloat64, math.MaxFloat64
+	minX, maxX := -math.MaxFloat32, math.MaxFloat32
+	minY, maxY := -math.MaxFloat32, math.MaxFloat32
 	proj, ok := geoprojbase.Projections[srid]
 	if ok {
 		minX, maxX = proj.Bounds.MinX, proj.Bounds.MaxX

